packages/api/src/Base.ts

```diff
@@ -532,6 +532,25 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
     decorated.key = (arg?: CodecArg): string =>
       u8aToHex(compactStripLength(method(arg))[1]);
 
+    // Linked Map support
+
+    if (method.headKey) {
+      decorated.head = (): C =>
+        onCall(
+          (arg: CodecArg) => this._rpcRx.state
+            .getStorage(arg)
```

Comment:
I was trying to implement this `subscribeStorage` but can't get it right.
It needs firstly get storage with head key, which will return the first key, and can be used to fetch first entry

Reply:
pleayed with your branch a bit, it works ok for me with `subscribeStorage` - just be aware that it returns an array, so you just need to do something like -

```
        onCall(
          (arg: CodecArg) => this._rpcRx.state
            .subscribeStorage([arg])
            .pipe(
              tap((value) => console.error('head', value)),
              // grab first value of the array 
              switchMap(([key]) => this._rpcRx.state.getStorage([method, key])),
              tap((value) => console.error('first', value))
            ),
          [method.headKey]
        ) as C;
```